sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/ExtractKeyPhraseClient.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics;
+
+import com.azure.ai.textanalytics.implementation.TextAnalyticsClientImpl;
+import com.azure.ai.textanalytics.implementation.models.DocumentError;
+import com.azure.ai.textanalytics.implementation.models.DocumentKeyPhrases;
+import com.azure.ai.textanalytics.implementation.models.KeyPhraseResult;
+import com.azure.ai.textanalytics.implementation.models.MultiLanguageBatchInput;
+import com.azure.ai.textanalytics.models.DocumentResultCollection;
+import com.azure.ai.textanalytics.models.ExtractKeyPhraseResult;
+import com.azure.ai.textanalytics.models.TextAnalyticsRequestOptions;
+import com.azure.ai.textanalytics.models.TextDocumentInput;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static com.azure.ai.textanalytics.Transforms.mapByIndex;
+import static com.azure.ai.textanalytics.Transforms.toBatchStatistics;
+import static com.azure.ai.textanalytics.Transforms.toMultiLanguageInput;
+import static com.azure.ai.textanalytics.Transforms.toTextAnalyticsError;
+import static com.azure.ai.textanalytics.Transforms.toTextDocumentStatistics;
+import static com.azure.core.util.FluxUtil.monoError;
+
+/**
+ * Helper class for managing extract keyphrase endpoint.
+ */
+class ExtractKeyPhraseClient {
+    private final ClientLogger logger;
+    private final TextAnalyticsClientImpl service;
+
+    /**
+     * Create a {@code ExtractKeyPhraseClient} that sends requests to the Text Analytics services's extract keyphrase
+     * endpoint.
+     *
+     * @param service The proxy service used to perform REST calls.
+     * @param logger The logger for the {@link TextAnalyticsAsyncClient} class.
+     */
+    ExtractKeyPhraseClient(TextAnalyticsClientImpl service, ClientLogger logger) {
+        this.logger = logger;
+        this.service = service;
+    }
+
+    Mono<Response<ExtractKeyPhraseResult>> extractKeyPhrasesWithResponse(String text, String language,
+        Context context) {
+        Objects.requireNonNull(text, "'text' cannot be null.");
```

Comment:
I am assuming that the public async client has all-containing try-catch block that catching this NPE and converts it to mono error. If not, ensure that these exceptions are propagated through the reactor error channel.

Reply:
yes, all the public async calls have wrapping to return a mono error.